cl/people_db/filters.py

```diff
@@ -284,6 +313,19 @@ class AttorneyFilter(NoEmptyFilterSet):
         queryset=Party.objects.all(),
         distinct=True,
     )
+    filter_nested_results = filters.BooleanFilter(
+        field_name="roles", method="filter_roles"
+    )
+
+    # Attributes for the mixin
+    # **Important:** Keep this mapping up-to-date with any changes to
+    # RelatedFilters or custom labels in this class to avoid unexpected
+    # behavior.
+    join_table_cleanup_mapping = {
+        "parties_represented__docket": "docket",
+        "parties_represented__attorney": "attorney",
```

Comment:
Should we include a mapping to `role` to?

`"parties_represented__role": "role"`

So role also available for filtering the nested results?

Reply:
[this comment](https://github.com/freelawproject/courtlistener/pull/4729/files#r1862587924) is also applicable here. To enable `role` lookup, we would need to update the class used in the `RelatedFilter` definition (specifically, the `PartyFilter` class in this instance)